### ISSUE TITLE
Making Hassio & Docker the only advertised installation methods

### ISSUE THREAD

**Issue by frenck** [home-assistant/architecture#126]
Up front, this request is a little out of emotional handling (let's call my personal anger/frustration). Also, I realize this is not really an architectural thing, but since I was unsure where to put it, I thought, this repo might be the best match. Mainly because many valuable conversations are going on in this repository.

**⚠️ This openings post contains MY opinion (as pointed out by @DubhAd)**

## That being said, let me do a TL;DR

> Suggestion for moving everything venv/manual (installation) instructions into the developer documentation. Drop Hassbian and other installation methods. Only advertise Docker & Hass.io installation methods on our main website.

Please, don't explode on this tl;dr right away, I've wrote a detailed and rational explaination below.

## Problem

We are currently trying to maintain everything focussed at mainly 3 installation types: The manual/venv one, Docker & Hass.io. (We do have multiple others... but let us keep those out of scope for this story).

This causes issues on our website, documentation, and confusion in our user base. Some problems with this, but not limited to these: 

- Our website has multiple "methods" to maintain.
- The documentation is fragmented and polluted with comment/notices and additional instructions of different platforms. Some examples include:
    - Differences in (configuration) path locations.
    - The need for installing additional packages (but often don't mention it is not needed on every platform, or just on a specific platform).
    - Instruction for executing command line tools that won't work on every platform.

## Results of the current strategy

- User's don't know what to install and what is the best method, which does not add to the initial user experience.
- Some users don't even know which platform they are using (or what it means).
- The fragmented/polluted documentation causes confusions, like:
    - Paths specified in the documentation do not match or work.
    - Following instructions on the documentation does not always work since it is installation type/platform depended. E.g., installing an additional Python or system package.
    - Users in Hass.io or Docker are trying to execute command lines listed in the documentation that does not work. Or in some cases, even command lines for very very specific setups are listed (e.g., Synology).
  - Heavy discussions among the user base on what the "recommended" installation type is, or even worse, what the best one is.
  - Misconceptions on Docker/Hass.io not being for "advanced" users, which are being carried by part of the community to new users as well.
- Community forums, chat and all is split. Users are bashing methods or giving non-matching answers to other users questions. Consing even more confusing among users.

## Suggestion

I want to suggest doing the following:

- Support only 3 installation methods:
    - venv/manual
    - Docker
    - Hass.io
- The venv/manual installation type needs to be targeted at "developers" (not "advanced" users).
- Move the venv/manual installation type into the developers documentation, since that is the target.
- Redirect/link users looking for the venv/manual method from our website to the developer docs.
- Drop Hassbian (a big one, but is simply does not fit this suggestion).
- Advertise Docker & Hass.io as the only installation methods on our homepage.
- Consider removing all the other installation methods OR make a specific corner for them with a notice on these are being community contributed methods and not officially supported.
- Write up proper Docker documentation, and maybe consider shipping Docker compose files in a separate repo.
- Major documentation cleanup.

## Motivation

We want to offer our userbase the best experience for a local and privacy based home automation platform. The current fragmentation does not allow us to make steps and/or get everything to a level of quality it should be for both the experience of installation, using and configuring, which includes our website and documentation. This hurts current and new user base and above all causes friction between users in our userbase time and time again.

Nevertheless, this is an open source project and we do largely rely on developers contributing code and fixes by donating they spare free time and knowledge. Therefore, I want to **strongly clarify** that I'm not suggesting on dropping venv/manual support, but just suggest on moving it where it belongs: the developers.

Besides solving the [Results of the current strategy](#results-of-the-current-strategy), there are some more advantages.

Like Docker base systems (Docker & Hass.io) are shipping with a collection of dependencies inside the container, which are easier to control and quality ensured.

Finally, users usually talk about "advanced" users. IMHO there are just contributing developers and users. There is almost nothing (maybe there are some edge cases I don't know?) that can't be done on Docker. So saying Docker/Hass.io is not for advanced users, is just plain old bullstuff, this misconception needs to be put to rest as well.

## Shockwave

The community will be experiencing a shock, and there will be quite some users that will go into rage mode. I do however think this will be a one-time painful chapter and we'll be able to move forward as an army afterwards.

## Clarification / Q&A

_(FYI: I've added this chapter later)_

Based on Discord discussions and feedback, I've added this section. Just to clarify things.

_"Frenck are you suggesting on..."_:

- _*"doing Hass.io only?"*_
    > No definitly not. Please god no. Docker & Hass.io. Those are 2 different things and covers about every single possible setup. I somehow feel like people do not realize Docker & Hass.io are different things (althought Hass.io of course uses the Docker engine).

- _*"Dropping venv support?"*_
    > Hell no! That is how we develop! This will **NEVER** go away. I'm just suggesting to move that part to the developers website and support it through that. If you think you should run it (eventhough you are not developing), you still can. And there is nothing wrong with that.

- _*"leaving our advanced users in the dark?"*_:
    > No. I'm not sure what an advanced user is to be honest. Everything can be done on Docker & Hass.io. We have users and developers in my view. I'm suggesting on making that separation more clear. If you are still stuck in the tought that you can't do everything with Hass.io or a Docker setup, you should try it once, since both methods are really "advanced" and mature.

- _*"So where just going to say F you to all other installation methods?"*_
   > No, I've suggested on either removing or listing them separately as "community" installations methods. Either way, Docker is supported by many many systems (Synology, QNap, Unraid, promox, freenas). Which IMHO is a better method compared to the Synology Python upgrade "hack" that recently is being added to our documentation.

- _*"Devs get venv, everybody else can leave?"*_
   > Really? It is an open source project, if you want to be running it as a developer, without developing, that is fine.

- _*"Dropping Docker as the next step, right?"*_
    > No! That would be a really bad thing. I love Docker and Hass.io is not for everybody.

## Blue pill or red pill

_(FYI: I've added this chapter later)_

This is issue is might be causing some major discussions, and I'm pretty sure we will not receive consensus. So I guess it eventually will be up to @balloob to decide.

Nevertheless, valuable and constructive input from everybody is helpful!

Kind regards,

Frenck

**Comment by cgarwood** [comment]
I agree we have too many "official" installation methods. Though I'm not completely sold on only advertising docker/hass.io and burying manual/venv behind the dev docs. I'd almost prefer we advertise hass.io and manual/venv as the two official methods.

**Comment by frenck** [comment]
@cgarwood Main issue with that, it that those are miles apart and will not deal with the fragmented documentation, website and user base.

**Comment by andrey-git** [comment]
Here is an "advanced user" use case. Is it supported by docker? (IIUC it is not supported by hassio)

Rpi with HA + other external services like Telegram bot.

**Comment by frenck** [comment]
@andrey-git Well, guess that is a misconception we can take care of right now. That will work, bot Docker & Hass.io (unless something recently broke?).

**Comment by pvizeli** [comment]
@andrey-git the library could also work with hass.io, but I had no time and hardware to fix that issue on build level. We work on a system they allow modifying the container on constrain time with a simple config. Telegram bot works for me on my installation.

**Comment by DubhAd** [comment]
So, shortest version is - great way to piss off a large slice of the user base. _**particularly**_ after it's been publicly (and privately) said by @balloob and others that there would be no Hass.io or the highway approach.

I get where this is coming from, but, it's a little late in the game without splitting and damaging the community. Lots of us (particularly the supporters) pre-date either Hass.io or Docker as options. Shutting us out (yes, that's what this is), makes us second class citizens overnight. Then it's just a matter of time until we're not supported any more, and we get to either rebuild everything based on Hass.io, or move elsewhere.

Then there's also the nagging concern that if the door was closed on Hassbian, what future does Docker have...

So, it makes sense, but you run the very significant risk of losing most of the active support base if you handle it the way you're proposing ;)

**Comment by frenck** [comment]
> I get where this is coming from, but, it's a little late in the game without splitting and damaging the community.

I agree, hence I think we should move, because the problem will grow and will become even bigger damage in the future.

> Shutting us out (yes, that's what this is), makes us second class citizens overnight. Then it's just a matter of time until we're not supported any more, and we get to either rebuild everything based on Hass.io, or move elsewhere.

No I'm not suggesting Hass.io only. That is not something I would want. I'm suggestion Docker & Hass.io.

Secondly, not supporting venv does not make sense, this is the way the system is developed...

> Then there's also the nagging concern that if the door was closed on Hassbian, what future does Docker have...

I don't get that one, Docker is a perfectly valid approach and support by many systems (e.g., Synology, Unraid, and a lot more).

> So, it makes sense, but you run the very significant risk of losing most of the active support base if you handle it the way you're proposing ;)

The current situation does harm as well imho.

**Comment by dgomes** [comment]
<rant>
This looks like a Linux distro war
</rant>

The confusion comes from having a single site for everything, it was bound to happen that some users feel 2nd/3rd class. It's always strange/complicated to explain a newbie the difference between hass.io and HA. The current state of affairs is no good, and I therefore agree with @frenck, on the other hand it is not very wise decision to make a top-down decision on what should be supported.

The solomonic decision would be to remove Hass.io, Docker, Venv and the likes from the main site and have different sites and projects for each of them. Let the community sort out themselves, just as Linux is packaged by different distributions, instead of privileging a given environment.

**Comment by DubhAd** [comment]
> > Shutting us out (yes, that's what this is), makes us second class citizens overnight. Then it's just a matter of time until we're not supported any more, and we get to either rebuild everything based on Hass.io, or move elsewhere.
> 
> No I'm not suggesting Hass.io only. That is not something I would **not** want. I'm suggesting Docker & Hass.io.
>
> Secondly, not supporting venv does not make sense, this is the way the system is developed... Hence moving that part into the developer docs (where is already kinda is already, duplicate).
> 
> > Then there's also the nagging concern that if the door was closed on Hassbian, what future does Docker have...
> 
> I don't get that one, Docker is a perfectly valid approach and support by many systems (e.g., Synology, Unraid, and a lot more).

The door is being closed on anything other than the two approved install methods. Why not reduce that in the future? That's the _obvious_ next step ;) Remember, this isn't just logic, this is about communication, history, and perception.

> > So, it makes sense, but you run the very significant risk of losing most of the active support base if you handle it the way you're proposing ;)
> 
> The current situation does harm as well imho.

Ok, bye. Have fun 👋

(Seriously, that came across as _don't care_)

**Comment by frenck** [comment]
I've updated the initial post with a "Clarification / Q&A" section to make my intentions more clear.

**Comment by DubhAd** [comment]
> * Those are 2 different things and covers about every single possible setup

In *your* opinion ;)

**Comment by frenck** [comment]
Yes, @DubhAd, absolutely. Added a disclaimer to the opening post to reflect that.

**Comment by fabaff** [comment]
First, we decided a while ago that we are focusing on Hass.io as the primary installation method. I don't see a point to start this discussion all over again but now for the advanced methods. Especially as all other methods are already and clearly marked as "advanced" in the documentation.

Docker is nice but as long as there are issues with the packages which the distributions provide and the users are forced to use `docker-ce`, the Docker way is just another advanced method. Also, using Docker adds an additional layer which is not really needed by everyone. 

Using a `venv` is definitely not a developer-only thing. It's pretty much the default way one should run Python tools if they are not provided by the distributions. 

Well, the documentation could be better in focusing on the primary installation method and the "hacks" for different platforms should be reviewed.

**Comment by OttoWinter** [comment]
I agree with focusing on hass.io as installation method (as has been decided before), but I don't like the idea of burying the pip installation in the dev docs. I'll give some of the experience I had with installing HA as reasons:

# Reason 1: Hass.io/Docker is not fully ready yet

First, I think the HassOS and docker installations are not there yet for the UX. Yes, they both have great potential for ease of installation, but the devil is in the details. Some time ago, I wanted to set up a RPi (3B) with a HA install as a temporary solution for my new home.

## Attempt 1: HassOS

Of course the first thing I tried was that shiny new HassOS. I followed the [getting started guide](https://www.home-assistant.io/getting-started/) first.

- Step 1: Downloaded and flashed base image.
- Step 2: WiFi config; I only have WiFi in my room so that was the preferred method.

  - First problem was finding a USB stick I could format with one of the supported file systems. I only had encrypted ext4 and APFS sticks around (all with data I didn't want to erase), so I had to use a microSD card with an adapter, ok fine.
  - Formatted it with ExFAT, created the my-network file. Problems:
  
    1. We're linking to GH **from the getting started guide** - This seems terribly messy.
    2. The [network docs](https://github.com/home-assistant/hassos/blob/dev/Documentation/network.md) are not too detailed and I think especially RPi beginners would struggle with this step.
    
  - Booted up RPi, waited 20 minutes or so ... but RPi does not connect to WiFi network :(
  - _Reads through [another document](https://github.com/home-assistant/hassos/blob/dev/Documentation/configuration.md) and finds out ExFAT is not supported_. Oh dumb me, I didn't know that.
  - _Recreates USB stick with FAT32_. Of course I didn't store the `my-network` file anywhere so I needed to rewrite it.
  - _Reboots RPi, waits a while_. Oh, of course the config is not read on second boot - or so it seemed.
  - _Reflashes microSD card and does everything from the beginning_. Smh didn't get WiFi to work despite numerous attempts, trying with AP from my phone etc. I just couldn't get it to work.
  - So in the end I managed to hook up an ethernet bridge from a windows machine. Problem was: that PC was using a cellular connection (long story), but the Hass.io install is *huge* and took ages to download.
  - Then the RPi worked. But of course I quickly ran out of storage on the 16GB RPi. Yes I know why Hass.io needs a lot of space (docker downloads), but I don't think something as lightweight as HA should require more than 16GB of storage for normal operation. I've installed tons of stuff on Raspbian with a 16GB card without a problem.

All in all that took 3-4 hours of troubleshooting, for something that should only take a couple minutes at most. If I didn't know about HA, I would have given up in the early stages - a getting started (especially the preferred method) should _not_ be this difficult.

Sure I did some mistakes while installing, but I don't think I'm alone with these problems.

**Time:** 3-4 hours

## Attempt 2: Docker

I like docker so I tried that installation method too. 

- First step was installing docker on RPis. I've only done so on 64 bit machines till now but I thought it wouldn't be that hard.
- After some quick googling. I found the ["convenience script"](https://docs.docker.com/install/linux/docker-ce/debian/#install-using-the-convenience-script). Well I don't like running arbitrary shellscripts, but hey I wanted to get it done quick.

  - Problem: The docs do not mention how to install docker (which I think they should)

- I ran the quick `docker run -it homeassistant/home-assistant` to get started - again downloading the image took an eternity. But of course that failed with format errors.
- Tried `homeassistant/raspberrypi3-homeassistant` - another eternity later it finally worked. Yay

Then I set up a docker compose file - and again forgot to change `homeassistant/home-assistant` for the RPi. But this installation method was much quicker compared to HassOS

**Time:** ~1.5 hour

## Attempt 3: PIP

For fun I also tried the good old `pip` - I was hoping this was at least not so bad of an UX as the other ones.

- Installed raspbian (again). Set up WiFi/SSH through the boot partition (as seen in numerous guides on internet) and had an SSH connection up and running in less than 15 minutes.
- Then, it was just a simple `apt install python3-pip` (which of course was pre-installed) and `pip3 install homeassistant`. Install took about a minute and another minute later my HA installation was up and running.
- After that of course a systemd script was required too but that was 2 minutes max with the numerous examples in the internet.

**Time:** 30 minutes

Yes it gets more difficult with virtualenvs and all, but on this machine I was going to run HA together with a quick jupyter server anyway. Personally speaking, I've never really had any package conflicts when not using a virtualenv, so I usually don't use them (except for development).

And yes, python installs can be horribly messy - `sudo pip` _or_ `pip` etc (I run everything as root, so no problem there). And yes we should add that to the docs (maybe even [this xkcd](https://xkcd.com/1987/)). I don't care if we put up huge blinking red banners warning users. For me at least `pip` was the quickest installation by far.

## Conclusion

Hass.io has a huge potential for ease of use, but I'm sorry to say that at least IMO the installation is not there yet. I'm sure we will solve these problems one by one in the future (somebody want to do an etcher fork with the ability to configure straight from UI? 😁)

The pip installation is especially simple when you already have an RPi up and running with other services. For example once I had a RPi musicbox set up - I wouldn't want to have to re-do the whole music box because I needed to flash another base OS.

(Also, I'm sure there are reasons for why the HassOS installation failed, but I'm not here to troubleshoot what failed.)

# Reason 2: Extensibility

In my mind, HA has always always been great because of its extensibility. If I want to add a custom component I can do so really easy - with other systems like OpenHAB that's just a nightmare.

But sometimes you need to install another system package. For example I once needed to have `git` in the HA install, which is not there in the default HA docker image. Overriding the HA Dockerfile for Hass.io is pretty hard right now (you'd essentially have to use the dev instructions).

For "pure" docker, it's a bit easier. Just extend the image with `FROM home-assistant/...`, run the build and you're good to go. Downside is with more complicated install scripts it often takes a couple of attempts to nail the `RUN` commands - re-running `docker build` on an RPi takes ages.

For `pip` installs it's the easiest. You only have to run `apt install git` on the host once and HA has access to `git` after that.

---------------------

So in conclusion: While the `pip` install might be the one we're getting most beginner questions about, I think the installation is still the simplest. And IMHO the time to get to a HA install should be as short as possible; an hour max. Yes we can add huge warning signs to the pip installation, but we should not remove it from the docs entirely.

**Comment by frenck** [comment]
I do agree with @ottowinter on some of those issues. Making choices can be accompanied by requirements that need to be solved in order to make a move (if that was decided).

For example, the HassOS/Hass.io flashing part, yes, please!  Make it easier. For example, the kerberos.io project has a fork of Etcher:

![](https://cdn.discordapp.com/attachments/470179143632289804/528683675525578762/unknown.png)

![](https://cdn.discordapp.com/attachments/470179143632289804/528683695067103242/unknown.png)

If we could find someone capable and willing to do something similar, that surely would help.

About Docker, that highlighted by me already. We need proper docs for that and provide a repo with a proper Docker compose and stuff. Another thing we could do is making use of the Docker manifests, which allows portability of the compose file across systems.


`pip` is easy (if you've set up your environment correctly). I agree, but we are developers or at least, we are IT guys (or maybe just one with an IT hobby or well, we do something with computers). As @fabaff pointed out, that is the way Python tools are run. My concern is that from a "general" user perspective, they actually don't care about that, that much.


Concerning this subject has been discussed before, true. I've seen flakes of it in many places. Make the discussion central, public and logged looked like a good idea to me. Whatever has been discussed in the past, I do think we have an issue.

Split community, confused and lost users, fragmented documentation, ... (well repeating my OP). So IMHO the issue at hand hasn't been solved in the last iteration.

**Comment by DubhAd** [comment]
> The documentation is fragmented and polluted with comment/notices and additional instructions of different platforms.

Moving to Hass.io *and* Docker doesn't solve this. Hass.io uses can install an add-on, Docker users can't.  The next obviously step, as I've said, is to reduce that to just Hass.io, then the docs will be totally consistent. Solving a documentation problem by reducing choice is an odd solution.

Why not simply tackle *that* problem?

I'd also point out that the Data Science thread that clearly kicked off the writing of this issue, also indicates the strength of feeling in the community - on all sides. Yes, some people didn't so much step over the line as take a running jump over it, but there are valid points (again, on all sides) in there. I'd TL/DR those as:

* Hass.io increases the potential user base, this is good
* The other install methods _overlap_ with that user base, but just as there are many Hass.io users who'd not consider those methods as sensible _for them_, so it works the other way around

> About Docker, that highlighted by me already. We need proper docs for that and provide a repo with a proper Docker compose and stuff. Another thing we could do is making use of the Docker manifests, which allows portability of the compose file across systems.

So... the docs need work 🤷‍♂️ That's not exactly news 😉 Reducing choice doesn't solve that problem unless you reduce that to no choice (which still doesn't solve the problem, just reduces it a tiny bit).

I've had _many_ people tell me in Discord that the OpenHAB docs are much more accessible than ours. Maybe we should be learning from what they get right.



> Well, the documentation could be better in focusing on the primary installation method and the "hacks" for different platforms should be reviewed.

That IMO is the most sensible solution. We promote Hass.io's instructions, and move the _on Debian..._ instructions further down. That avoids splitting the community, and tackles the actual problem.

**Comment by Vasiley** [comment]
My two cents..

HA will always be python and advanced users like this and don't mind that it takes some extra configuration. It always helps to learn something hands on. Hassio has its place and it does a great job. I would think that you would test functions in its most basic way and then move it to a prepackaged format for use ie. Hassio.  I think support would be much easier if a roll out of a feature included those that are most likely to be able to work out bugs and report back. As far as the doc's it could be hassio on top and advanced on the bottom and i dont see any problem maintaining that as I and others i am sure would help.

P.S.S.
 I dont think there needs to be any anger just move forward with the core and continue to improve....

Thanks in advance
Vasiley

**Comment by Landrash** [comment]
_My point of view._

We (I haven't contributed much lately so _we_  as the project in general) have a documentation problem. 
The advanced user base shouldn't be pushed into looking developer documentation for a simple answer since an advanced user isn't necessarily a developer. Moving the bar to the mythical new user who never touched a computer isn't necessarily where the documentation should be aimed either.
I'm not sure there's a simple answer but I'm fairly sure pushing the advanced user further away is it.

**Point by point**
>User's don't know what to install and what is the best method, which does not add to the initial user experience. Some users don't even know which platform they are using (or what it means).

A beter landing page for the new user is the solution to the first half.
The second part is harder since teaching a user about computers and architecture really should be in the scope of HA.

>The fragmented/polluted documentation causes confusions, like:  Paths specified in the documentation do not match or work.

Maintainers for component documentation is most likely what's necessarily here.

>Following instructions on the documentation does not always work since it is installation type/platform depended. E.g., installing an additional Python or system package.

If this is the case that's an issue with documentation needing to be updated and not dropping support of a big group of people.

> Users in Hass.io or Docker are trying to execute command lines listed in the documentation that does not work. Or in some cases, even command lines for very very specific setups are listed (e.g., Synology).

Then there's a need to educate the users that they are on limited platform and some of these options won't work.

> Heavy discussions among the user base on what the "recommended" installation type is, or even worse, what the best one is.

This discussion isn't something bad and will always be a part of an open source community.

> Misconceptions on Docker/Hass.io not being for "advanced" users, which are being carried by part of the community to new users as well.

Here we can have a totally separate discussion. I agree since this group of users are either on the spectrum of beginners on the platform or seasoned veterans with docker.

> Community forums, chat and all is split. Users are bashing methods or giving non-matching answers to other users questions. Consing even more confusing among users.

You can't control speech and opinion and this would most likely never stop.. Tag posts with official answer and link to documentation where this is a problem. Removing the options of other install options won't stop this.

Cleaning up the configs and adding maintainers for certain installation options may be a solution to keep documentation up to date.

> Drop Hassbian (a big one, but is simply does not fit this suggestion).

**That one you could have had a private discussion if you wanted to talk about it before posting it like this.**
Home Assistant doesn't support it currently more than the general mention in the documentation. It's intended for those who want's a simple install option while maintaining a basic Raspbian based install.
It currently installs as a basic venv installation without any extras.


There's already the problem with hampering the usage of config files and pushing for a UI only approach so doing this will just push the advanced users further away.

**Comment by frenck** [comment]
Thanks for your response @Landrash!

Just one thing I do not understand is:

> That one you could have had a private discussion if you wanted to talk about it before posting it like this.

I thought this was an open source project. Sorry, me discussing things in the open.

**Comment by balloob** [comment]
I'm busy packing all my stuff and moving between continents, but just a few small thoughts and tidbits.

1. Besides getting the base system and component's dependencies up and running, there shouldn't be many things different across documentation for Home Assistant and integrations?
1. I think that we should add a more dedicated "marketing" page to explain what Hass.io is. There is a lot of misunderstandings floating around, including the one that it is not for advanced users. If you click on our getting started, even our link to alternative installation methods says that advanced users should check there 🙄([my fault btw](https://github.com/home-assistant/home-assistant.io/pull/4194/files#diff-2fda6257ec23f5022a1292845530041fR14))
2. Which installation instructions in our alternative installs section does not has the user ending up with Home Assistant in a venv that they then have to maintain? (besides Docker)
3. Maybe we can have something like an "installation method" dropdown on the top right of the site, and when people pick one, pages can enable links or extra docs for that installation type. Could be as simple powered by HTML & CSS:
    ```html
    <div class='device-specific docker'>Bla die bla</div>
    ```
    However, I also like the idea if such things for obscure methods could be maintained outside of our official repo's, as there are an unlimited amount of installation methods and putting the burden of reviewing changes, bug reports on outdated info etc on our docs maintainers doesn't scale (as we also see with our Python repo)

**Comment by Landrash** [comment]
@frenck
> I thought this was an open source project. Sorry, me discussing things in the open.

It is in general common courtesy to at least mention it to the folks maintaining it that you want the project to drop it. If anything it annoyed me personally since I and @luddeus maintain and try to keep it a viable option. It's an open source project but the focus has had a strange shift to ignore part of the advanced user base in favour of growth.

This is partly why so many get annoyed. Suggestions and changes are introduced and part of the community doesn't feel like they get listened to. 

Since the accepted way for change should and might be an Request for Changed it might be time that the part of the community that feel ignored and forgotten start writing a few.

**Comment by DubhAd** [comment]
> This is partly why so many get annoyed. Suggestions and changes are introduced and part of the community doesn't feel like they get listened to.

I've commented elsewhere on this a few times (including a few long chats with @dale3h on Discord), and possible banged on quite a bit more in other channels too ;) The project has grown to the point where _effective_ communication is a problem. Introducing some more formal method for discussing functional changes _and getting feedback in advance_ is needed.

I'll pick the plan to drop column settings in the glance card in Lovelace as an example of what could have impacted lots of folks negatively, just because there wasn't any public discussion.

Lots of people have built their automation environment around Home Assistant because it's flexible, powerful, and supports many ways of getting started. It seems like every week another person joins who says that they left some other platform for HA because of those things (plus all those people that never say). Many of them make the same comments though - the documentation is harder to follow ([example here](https://discordapp.com/channels/330944238910963714/330944238910963714/528875901556097024)), and the communication can be confusing.

We can fix both of those **together**, or we can let the few rude voices cause us to argue and fight, ultimately harming the whole community.

PS: In light of this thread I did go looking at the other open source options out there, to see how viable a switch might be. In terms of functionality Home Assistant is quite far ahead. We shouldn't let a few loudmouths who hate on Hass.io (or anything else for that matter) ruin this.

**Comment by frenck** [comment]
With all respect @Landrash, discussing things in the open is how an opensource project work imho. With everything that I'm suggesting in my OP, a lot of people are involved. Do you expect me to contact them each one personally?

> It's an open source project but the focus has had a strange shift to ignore part of the advanced user base in favour of growth.

:arrow_up: Hence me doing this open and public. I don't get why that is an issue?

Everybody is allowed to share opinions, so do I. If you prefer me to not do that, just say so.

**Comment by Landrash** [comment]
@DubhAd This seems like a good compromise.
 
Any component specific changes that are needed can easily be changed in the Hassbian image also if so preferred.
